src/main/java/org/spongepowered/common/data/manipulator/immutable/item/ImmutableSpongeBreakableData.java

```diff
@@ -0,0 +1,109 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.item;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Booleans;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableBreakableData;
+import org.spongepowered.api.data.manipulator.mutable.item.BreakableData;
+import org.spongepowered.api.data.value.immutable.ImmutableSetValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.item.SpongeBreakableData;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeSetValue;
+import org.spongepowered.common.util.GetterFunction;
+
+import java.util.Set;
+
+@NonnullByDefault
+public class ImmutableSpongeBreakableData extends AbstractImmutableData<ImmutableBreakableData, BreakableData> implements ImmutableBreakableData {
+    private final ImmutableSet<BlockType> breakable;
+
+    public ImmutableSpongeBreakableData() {
+        this(ImmutableSet.<BlockType>of());
+    }
+
+    public ImmutableSpongeBreakableData(Set<BlockType> breakable) {
+        super(ImmutableBreakableData.class);
+        registerGetters();
+        this.breakable = ImmutableSet.copyOf(breakable);
+    }
+
+    @Override
+    public int compareTo(ImmutableBreakableData o) {
+        return Booleans.compare(o.breakable().containsAll(this.breakable),
+                this.breakable.containsAll(o.breakable().get()));
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        Set<String> breakableIds = Sets.newHashSetWithExpectedSize(this.breakable.size());
+        for (BlockType breakable : this.breakable) {
+            breakableIds.add(breakable.getId());
+        }
+        return new MemoryDataContainer().set(Keys.BREAKABLE_BLOCK_TYPES.getQuery(), breakableIds);
+    }
+
+    @Override
+    public ImmutableSetValue<BlockType> breakable() {
+        return new ImmutableSpongeSetValue<BlockType>(Keys.BREAKABLE_BLOCK_TYPES, this.breakable);
```

Comment:
You might want to compute this value in the constructor, and just return that instance for this method.